integration-tests/src/test_loop/tests/epoch_sync.rs

```diff
@@ -221,4 +231,115 @@ fn test_epoch_sync_from_genesis() {
         .map(|s| s.to_string())
         .collect::<Vec<_>>()
     );
+
+    let mut stores = Vec::new();
+    for data in &node_datas {
+        stores.push(
+            test_loop
+                .data
+                .get(&data.client_sender.actor_handle())
+                .client
+                .chain
+                .chain_store
+                .store()
+                .clone(),
+        );
+    }
+
+    let tempdir = TestLoopEnv { test_loop, datas: node_datas, tempdir }
+        .shutdown_and_drain_remaining_events(Duration::seconds(5));
+
+    TestNetworkSetup { tempdir, genesis: setup.genesis, accounts: setup.accounts, stores }
+}
+
+// Test that a new node that only has genesis can use Epoch Sync to bring itself
+// up to date.
+#[test]
+fn test_epoch_sync_from_genesis() {
+    init_test_logger();
+    let setup = setup_initial_blockchain(4);
+    bootstrap_node_via_epoch_sync(setup, 0);
+}
+
+// Tests that after epoch syncing, we can use the new node to bootstrap another
+// node via epoch sync.
+#[test]
+fn test_epoch_sync_from_another_epoch_synced_node() {
```

Comment:
Wondering if it makes sense to add test where we
- setup_initial_blockchain
- bootstrap_node_via_epoch_sync for node 4 via 0
- Run through a couple more epochs (so that proof is old)
- bootstrap_node_via_epoch_sync for node 5 via 4

Meanwhile do sanity checks to see if the old proof is store on node 0, 4 (after first bootstrap) and new proof is stored on node 4 and 5?

Reply:
That actually *is* what is being done. For proof storage, the old proof is not stored on 0 (it never will be); the old proof is stored on 4 and new proof is stored on 5. I'll add sanity check that the proofs exist.